src/main/java/org/checkstyle/autofix/data/CheckstyleRecord.java

```diff
@@ -0,0 +1,147 @@
+package org.checkstyle.autofix.data;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.thymeleaf.util.StringUtils;
+
+/**
+ * POJO, maps into single "error" tag of the XML.
+ */
+public final class CheckstyleRecord {
```

Comment:
Check fields and methods in this class, some of them not even used. Why did you add them?